packages/block-library/src/image/index.php

```diff
@@ -27,16 +28,67 @@ function render_block_core_image( $attributes, $content ) {
 		// which now wraps Image Blocks within innerBlocks.
 		// The data-id attribute is added in a core/gallery `render_block_data` hook.
 		$processor->set_attribute( 'data-id', $attributes['data-id'] );
+	}
+
+	$link_destination = isset( $attributes['linkDestination'] ) ? $attributes['linkDestination'] : 'none';
+	$enable_lightbox  = isset( $attributes['enableLightbox'] ) ? $attributes['enableLightbox'] : false;
+
+	if ( 'none' === $link_destination && $enable_lightbox ) {
+
+		$aria_label = 'Open image lightbox';
+		if ( $processor->get_attribute( 'alt' ) ) {
+			$aria_label .= ' - ' . $processor->get_attribute( 'alt' );
+		}
+
+		$background_color  = wp_get_global_styles( array( 'color', 'background' ) );
+		$close_button_icon = '<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" width="30" height="30" aria-hidden="true" focusable="false"><path d="M13 11.8l6.1-6.3-1-1-6.1 6.2-6.1-6.2-1 1 6.1 6.3-6.5 6.7 1 1 6.5-6.6 6.5 6.6 1-1z"></path></svg>';
+
 		$content = $processor->get_updated_html();
+
+		return
+			<<<HTML
+				<div class="wp-lightbox-container"
+					data-wp-island=""
+					data-wp-context='{ "core": { "image": { "initialized": false, "lightboxEnabled": false, "lastFocusedElement": null } } }'>
+						<button aria-haspopup="dialog" aria-label="$aria_label" data-wp-on.click="actions.core.image.showLightbox">
+							$content
+						</button>
+						<div data-wp-body="" class="wp-lightbox-overlay"
+							aria-role="dialog"
+							aria-hidden="true"
+							data-wp-class.initialized="context.core.image.initialized"
+							data-wp-class.active="context.core.image.lightboxEnabled"
+							data-wp-bind.aria-hidden="!context.core.image.lightboxEnabled"
+							data-wp-effect="effects.core.image.initLightbox"
+							data-wp-on.keydown="actions.core.image.handleKeydown"
+							data-wp-on.mousewheel="actions.core.image.hideLightbox"
+							data-wp-on.click="actions.core.image.hideLightbox"
+							>
```

Comment:
The dialog should have a label of some kind. Normally this is conveyed via aria-labelledby pointing to content inside the modal; e.g., the image itself.

Reply:
Ok this has been updated